string.c

```diff
@@ -9159,6 +9159,73 @@ rb_str_end_with(int argc, VALUE *argv, VALUE str)
     return Qfalse;
 }
 
+/*
+ *  call-seq:
+ *     str.delete_prefix!(substr)   -> self or nil
+ *
+ *  Deletes leading <code>substr</code> from <i>str</i>, returning
+ *  <code>nil</code> if no change was made.
+ *
+ *     "hello".delete_prefix!("hel")  #=> lo
+ *     "hello".delete_prefix!("llo")  #=> nil
+ */
+
+static VALUE
+rb_str_delete_prefix_bang(VALUE str, VALUE substr)
+{
+    long loffset;
+
+    str_modify_keep_cr(str);
+    if (rb_str_start_with(1, &substr, str) == Qtrue) {
+        loffset = RSTRING_LEN(substr);
+    }
+    else {
+        return Qnil;
```

Comment:
I'd prefer this and the below to be written as`if (...) return Qnil;` to reduce indented code blocks.